src/development/platform-integration/android/restore-state-android.md

```diff
@@ -0,0 +1,212 @@
+  _Instance state_ or
+  (also called _short-term_ or _ephemeral_ state),
+  includes unsubmitted form field values, the currently
+  selected tab, and so on. On Android, this is
+  limited to 1 MB and, if the app exceeds this,
+  it crashes with a `TransactionTooLargeException`
+  error in the native code.
+  To learn more about short term and long term state,
+  check out [Differentiate between ephemeral state
+  and app state][state].
+{{site.alert.end}}
+
+[state]: {{site.url}}/development/data-and-backend/state-mgmt/ephemeral-vs-app
+
+The `RestorationManager` manages the restoration
+data, which is serialized to a tree of [`RestorationBucket`][],
+that are synchronized with the Flutter engine.
+A `RestorationBucket` contains a key/value pair
+of state information that you want to save. The key is a
+[`restorationId`][] that is unique to that `RestorationScope`.
+
+Flutter automatically creates a
+[`RestorationScope`][] for each route in your
+app to track the `RestorationBucket`s in that scope.
+In general, you shouldn't need to worry about
+this object _unless_ the scope has two (or more) buckets with
+the same `restorationId`. If that occurs,
+create a new `RestorationScope` to ensure
+that each ID is unique to its scope.
+Each scope is assigned a unique [`restorationScopeId`][].
+
+You must decide what state you want to save and restore,
+but you can only save specific types to a `RestorationBucket`,
+namely: `null`, `bool`, `int`, `double`, `String`, `Uint8List`
+(and other typed data), `List`, `Map`, and child `RestorationBucket`s.
+
+{{site.alert.note}}
+  The code in this page was taken from [VeggieSeasons][].
+  VeggieSeasons is a sample app written for iOS that uses Cupertino
+  widgets. An iOS app requires [a bit of extra setup][] in Xcode, but
+  the restoration classes otherwise work the same on both iOS and Android.
+{{site.alert.end}}
+
+
+[a bit of extra setup]: {{site.api}}/flutter/services/RestorationManager-class.html#state-restoration-on-ios
+[`restorationId`]: {{site.api}}/flutter/widgets/RestorationScope/restorationId.html
+[`restorationScopeId`]: {{site.api}}/flutter/widgets/RestorationScope/restorationScopeId.html
+[`RestorationScope`]: {{site.api}}/flutter/widgets/RestorationScope-class.html
+[VeggieSeasons]: {{site.github}}/flutter/samples/tree/master/veggieseasons
+
+## Step 1: Enabling state restoration
+
+Most every Flutter app contains a `main` method that calls the
+[`runApp`][] function. You can use `runApp`
+to set up the framework and the bindings that bind
+the Dart code to the native Flutter engine.
+Set up the `RestorationManager` in the `runApp` function.
+For example:
+
+```dart
+void main() {
+  /...
+
+  runApp(
+    const RootRestorationScope(
+      restorationId: 'root',
+      child: VeggieApp(),
+    ),
+  );
+}
+```
+
+QUESTION for Michael: I thought that Flutter automatically created
+  the Scope for every route. But this app manually
+  creats a Scope for the page. ?? (xxx)
```

Comment:
In most cases, you can just specify a restorationScopeID on MaterialApp/CupertinoApp/WidgetsApp to enable state restoration: https://master-api.flutter.dev/flutter/material/MaterialApp/restorationScopeId.html. This will automatically insert a RootRestorationScope for you. However, the VeggieSeasons app has restorable state that lives ABOVE the CupertinoApp. That is why it manually inserts a RootRestorationScope.